payjoin/tests/integration.rs

```diff
@@ -222,6 +222,123 @@ mod integration {
+                let pj_uri = Uri::from_str(&session.pj_uri().to_string())
+                    .map_err(|e| e.to_string())?
+                    .assume_checked()
+                    .check_pj_supported()
+                    .map_err(|e| e.to_string())?;
+                let psbt = build_sweep_psbt(&sender, &pj_uri)?;
+                let new_sender =
+                    SenderBuilder::new(psbt, pj_uri).build_recommended(FeeRate::BROADCAST_MIN)?;
+                let storage_token =
+                    new_sender.persist(&mut NoopPersister).map_err(|e| e.to_string())?;
+                let req_ctx =
+                    Sender::load(storage_token, &NoopPersister).map_err(|e| e.to_string())?;
+                let (Request { url, body, content_type, .. }, _send_ctx) =
+                    req_ctx.extract_v2(ohttp_relay.to_owned())?;
+                let response = agent
+                    .post(url.clone())
+                    .header("Content-Type", content_type)
+                    .body(body.clone())
+                    .send()
+                    .await?;
+                log::info!("Response: {:#?}", &response);
+                assert!(response.status().is_success(), "error response: {}", response.status());
+                // POST Original PSBT
+
+                // **********************
+                // Inside the Receiver:
+
+                // GET fallback psbt
+                let (req, ctx) = session.extract_req(&ohttp_relay)?;
+                let response = agent
+                    .post(req.url)
+                    .header("Content-Type", req.content_type)
+                    .body(req.body)
+                    .send()
+                    .await?;
+                // POST payjoin
+                let mut proposal = session
+                    .process_res(response.bytes().await?.to_vec().as_slice(), ctx)?
+                    .expect("proposal should exist");
+                // Generate replyable error
+                let server_error = || {
+                    proposal
+                        .clone()
+                        .clone()
+                        .check_broadcast_suitability(None, |_| Err("mock error".into()))
+                        .expect_err("expected broadcast suitability check to fail")
+                };
+
+                let (err_req, err_ctx) =
+                    proposal.clone().extract_err_req(&server_error().into(), ohttp_relay)?;
+                let err_response = agent
+                    .post(err_req.url)
+                    .header("Content-Type", err_req.content_type)
+                    .body(err_req.body)
+                    .send()
+                    .await?;
+
+                let err_bytes = match err_response.bytes().await {
+                    Ok(bytes) => bytes,
+                    Err(e) => panic!("Failed to get error response bytes: {e}"),
+                };
```

Comment:
Could this be

```suggestion
                let err_bytes = err_response.bytes().await.expect("Failed to get error response bytes: {e}");
```

 or even

```suggestion
                let err_bytes = err_response.bytes().await?;
```

?